### DISCUSSION TITLE
Keep forward pass for calculating the grad later

### DISCUSSION THREAD

**Discussion by JGameCreation** [google/flax#2252] [Q&A] [answered]
Hi! I'm doing multiple forward passes through my Flax neural network and later want to calculate the gradient for one of them. However, I don't know for which one beforehand. Is there a way I can keep my information from the forward pass to make calculating that gradient later more efficient?
Best, Jannis

**Comment by jheek** [comment, marked as answer]
You can do this with [jax.vjp](https://jax.readthedocs.io/en/latest/_autosummary/jax.vjp.html?highlight=vjp#jax.vjp).

e.g.:
```python
loss, vjp_fn = jax.vjp(loss_fn, params, x)
# later...
grad = vjp_fn(1.)
```
